src/Model/Response/AsyncResponse.php

```diff
@@ -27,10 +31,27 @@ public function getContent(): string|iterable|object|null
         return $this->unwrap()->getContent();
     }
 
+    public function getRawResponse(): HttpResponse
+    {
+        return $this->response;
+    }
+
+    public function setRawResponse(HttpResponse $rawResponse): void
+    {
+        // Empty by design as the raw response is already set in the constructor and must only be set once
+        throw new RawResponseAlreadySet();
+    }
+
     public function unwrap(): ResponseInterface
     {
         if (!$this->isConverted) {
             $this->convertedResponse = $this->responseConverter->convert($this->response, $this->options);
+
+            if (null === $this->convertedResponse->getRawResponse()) {
+                // Fallback to set the raw response when it was not handled by the response converter itself
+                $this->convertedResponse->setRawResponse($this->response);
```

Comment:
this is currently the path for the most cases, right?

Reply:
Yeah. I just wanted to open up that people can do it also in the response converter for themself ... you never know.

Reply:
yup, fine, was just curious if i read it right